modules/core/src/main/java/org/apache/ignite/internal/processors/configuration/distributed/DistributedThinClientConfiguration.java

```diff
@@ -25,21 +25,36 @@
 
 import static org.apache.ignite.internal.cluster.DistributedConfigurationUtils.makeUpdateListener;
 import static org.apache.ignite.internal.processors.configuration.distributed.DistributedBooleanProperty.detachedBooleanProperty;
+import static org.apache.ignite.internal.processors.configuration.distributed.DistributedIntegerProperty.detachedIntegerProperty;
 
 /**
  * Thin client distributed configuration.
  */
 public class DistributedThinClientConfiguration {
-    /** */
-    private final IgniteLogger log;
+    /** Default connection count limit. */
+    public static final int DFLT_MAX_CONNECTIONS_PER_NODE = 0;
+
+    /** Message of thin client distributed config parameter was changed. */
+    private static final String PROPERTY_UPDATE_MESSAGE =
+            "ThinClientProperty parameter '%s' was changed from '%s' to '%s'";
 
     /** . */
     private final DistributedChangeableProperty<Boolean> showStackTrace =
-        detachedBooleanProperty("thinClientProperty.showStackTrace");
+            detachedBooleanProperty("thinClientProperty.showStackTrace");
 
-    /** Message of baseline auto-adjust parameter was changed. */
-    private static final String PROPERTY_UPDATE_MESSAGE =
-        "ThinClientProperty parameter '%s' was changed from '%s' to '%s'";
+    /**
+     * Maximum allowed number of active thin client connections per node.
+     * <p>
+     * Defaults to {@link #DFLT_MAX_CONNECTIONS_PER_NODE}.
+     * This applies to any connections that use thin client protocol: thin clients, ODBC, thin JDBC connections.
+     * The total number of all connections (ODBC+JDBC+thin client) can not exceed this limit.
+     * Zero or negative value means no limit.
+     */
+    private final DistributedChangeableProperty<Integer> maxConnectionsPerNode =
+            detachedIntegerProperty("maxConnectionsPerNode");
```

Comment:
Should we use `thinClientProperty.` prefix here as well? This name is a public API. It might be confusing, because it's not clear, what connection do we mean exactly

Reply:
So, `"thinClientProperty.maxConnectionsPerNode"`?

Reply:
It's going to be inconsistent with other thin client properties. How about `maxThinClientConnectionsPerNode`?